packages/quantic/force-app/main/default/lwc/quanticResultLink/__tests__/quanticResultLink.test.js

```diff
@@ -0,0 +1,105 @@
+// @ts-nocheck
+import '@testing-library/jest-dom';
+import * as mockHeadlessLoader from 'c/quanticHeadlessLoader';
+import { getNavigateCalledWith } from 'lightning/navigation';
+import { createElement } from 'lwc';
+import QuanticResultLink from '../quanticResultLink';
+import mockDefaultResult from './data/defaultResult.json';
+import mockKnowledgeArticleResult from './data/knowledgeArticleResult.json';
+import mockSalesforceResult from './data/salesforceResult.json';
+
+
+jest.mock('c/quanticHeadlessLoader');
+
+function createTestComponent(options) {
+  const element = createElement('c-quantic-result-link', {
+    is: QuanticResultLink,
+  });
+
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+// Helper function to mock headless for this suite of tests.
+function mockHeadless() {
+  jest.spyOn(mockHeadlessLoader, 'getHeadlessBundle').mockReturnValue(
+    new Promise((resolve) => {
+      resolve();
+    })
+  );
```

Comment:
The return value of `getHeadlessBundle` is not supposed to be a promise, it's supposed to return the following:
```
{
    buildInteractiveResult: jest.fn(),
}
```
```suggestion
  jest.spyOn(mockHeadlessLoader, 'getHeadlessBundle').mockReturnValue(
   {
    buildInteractiveResult: jest.fn(),
   }
  );
```